### PR TITLE
[site] optimize showcase images

### PR DESCRIPTION
**Pull request by benmccann**

### CODE REVIEW

File: sites/kit.svelte.dev/src/lib/Image.svelte [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<script>
+	export let src;
+	export let alt;
+	export let width = undefined;
+	export let height = undefined;
```
Comment (Rich-Harris) [inline sites/kit.svelte.dev/src/lib/Image.svelte:5]: we should add types here

File: sites/kit.svelte.dev/vite.config.js [hunk 2]
```diff
@@ -2,13 +2,29 @@ import { sveltekit } from '@sveltejs/kit/vite';
 import * as path from 'path';
 import { imagetools } from 'vite-imagetools';
 
+const supportedExtensions = ['png', 'jpg', 'jpeg'];
```
Comment (Rich-Harris) [inline sites/kit.svelte.dev/vite.config.js:5]:
```suggestion
import * as path from 'path';
import { imagetools } from 'vite-imagetools';

const supportedExtensions = ['.png', '.jpg', '.jpeg'];
```

File: sites/kit.svelte.dev/vite.config.js [hunk 3]
```diff
@@ -2,13 +2,29 @@ import { sveltekit } from '@sveltejs/kit/vite';
...
 const config = {
 	assetsInclude: ['**/*.vtt'],
 
 	logLevel: 'info',
 
-	plugins: [imagetools(), sveltekit()],
+	plugins: [
+		imagetools({
+			defaultDirectives: (url) => {
+				const extension = url.pathname.substring(url.pathname.lastIndexOf('.') + 1);
```
Comment (Rich-Harris) [inline sites/kit.svelte.dev/vite.config.js:16]:
```suggestion
				const extension = path.extname(url.pathname);
```

**Review by Rich-Harris — CHANGES_REQUESTED** [review]
if we do this then we probably want to have pngs rather than jpgs for the original screenshots, otherwise we're sacrificing quality needlessly

File: sites/kit.svelte.dev/src/lib/Image.svelte (hunk 1 above)
Comment (benmccann) [reply sites/kit.svelte.dev/src/lib/Image.svelte]: I've updated this as well as I can. I'm going to have to expose the `Picture` type from `vite-imagetools` before I can fully do it, but don't want to block on that

File: sites/kit.svelte.dev/vite.config.js [hunk 4]
```diff
@@ -2,13 +2,29 @@ import { sveltekit } from '@sveltejs/kit/vite';
...
 	logLevel: 'info',
 
-	plugins: [imagetools(), sveltekit()],
+	plugins: [
+		imagetools({
+			defaultDirectives: (url) => {
+				const extension = path.extname(url.pathname);
+				if (supportedExtensions.includes(extension)) {
+					return new URLSearchParams({
+						format: 'avif;webp;' + extension,
```
Comment (Rich-Harris) [inline sites/kit.svelte.dev/vite.config.js:19]: ah crap, i guess this should be this now? though it seems to work anyway...

```suggestion
						format: 'avif;webp;' + extension.slice(1),
```

Comment (Rich-Harris) [reply sites/kit.svelte.dev/vite.config.js:19]: gonna leave it as-is, if it ain't broke etc

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**